src/components/header/Header.tsx

```diff
@@ -1,13 +1,21 @@
-import { auth, signIn } from "@/auth/auth";
+'use client'
 
-export default async function Header() {
-  const session = await auth();
-  const user = session?.user;
+import {User} from '../../types/user';
+import {useState} from "react";
+import SearchInput from 'components/header/SearchInput';
+
+type Props = {
+  handleLogin: () => void;
+  user: User;
```

Comment:
We can update this to a boolean prop `isLoggedIn` as per my previous comment and delete the type definition since this component doesn't really use the properties of the User - it just needs to know that the user exists.
```suggestion
  isLoggedIn: boolean;
```